src/framework-dfp.c

```diff
@@ -186,6 +378,197 @@ void fcg_add_pairwise_edges(Graph *fcg, int v1, int v2, PlutoProg *prog, int *co
+                                        inter_scc_constraints->is_eq[row_offset+stmt2_offset+dim2] = 0;
+                                    }
+                                }
+                                /* Check if fusing ith dimesion of the source with ith dimension
+                                 * of the target is valid */
+
+                                prog->num_lp_calls ++;
+                                tstart = rtclock();
+                                sol = pluto_fusion_constraints_feasibility_solve(inter_scc_constraints, obj);
+                                prog->mipTime += rtclock()-tstart;
+
+                                /* If no solutions, then dimensions are not fusable. Add an edge in the conflict graph. */
+                                if (sol == NULL) {
+                                    IF_DEBUG(printf("Unable to fuse dimension %d of scc %d with dimension %d of scc %d \n",dim1,scc1 ,dim2 ,scc2););
+                                    IF_DEBUG(printf(" Adding edge %d to %d in fcg\n",scc1_fcg_offset+dim1,scc2_fcg_offset+dim2););
+                                    fcg->adj->val[scc1_fcg_offset+dim1][scc2_fcg_offset+dim2] = 1;
+                                } else {
+                                    IF_DEBUG(printf("Able to fuse dimension %d of scc %d with dimension %d of scc %d \n",dim1,scc1 ,dim2 ,scc2););
+                                    if (check_parallel) {
+                                        if (!is_lp_solution_parallel(sol,npar)) {
+                                            IF_DEBUG(printf("Adding Parallelism preventing edge %d to %d in fcg \n", scc1_fcg_offset+dim1, scc2_fcg_offset+dim2););
+                                            par_preventing_adj_mat->val[scc1_fcg_offset+dim1][scc2_fcg_offset+dim2] = 1;
+
+                                        }
+                                    }
+                                    free(sol);
+                                }
+                                /* Reset the lower bounds of dimensions of each statement in SCC2 */
+                                for (j=0; j<sccs[scc2].size; j++) {
+                                    stmt2 = sccs[scc2].vertices[j];
+                                    if (dim2<=stmts[stmt2]->dim_orig) {
+                                        stmt2_offset = npar+1+(nvar+1)*stmt2;
+                                        inter_scc_constraints->val[row_offset+stmt2_offset+dim2][CST_WIDTH-1] = 0;
+                                        inter_scc_constraints->is_eq[row_offset+stmt2_offset+dim2] = 1;
+                                    }
+                                }
+                            }
+                        }
+                        /* Set lb of dim1 of each statement in Scc 1 */
+                        for (i=0; i<sccs[scc1].size; i++) {
+                            stmt1 = sccs[scc1].vertices[i];
+                            if (dim1<=stmts[stmt1]->dim_orig) {
```

Comment:
Always leave spaces, i.e., if (dim1 <= stmts[...)  for readability. Also, better to rewrite as:
if (dim > stmts[stmt1]->dim_orig) continue;
// next three statements here

That way you have less indentation.